File: 项目/OC项目/爱鲜蜂(OC)/爱鲜蜂(OC)/Classes/Other/Extension/UIBarButtonItem+Extension.m
//
//  UIBarButtonItem+Extension.m
//  爱鲜蜂(OC)
//
//  Created by 金亮齐 on 2017/5/10.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "UIBarButtonItem+Extension.h"
#import "ItemCustomButton.h"

@implementation UIBarButtonItem (Button)
+ (instancetype)barButtonItem:(NSString *)title image:(NSString *)image target:(id)target action:(SEL)action type:(ItemButtonType)type{
    UIButton *button;
    if (type == LeftItemButton) {
        button = [ItemLeftButton buttonWithType:UIButtonTypeCustom];
    }else{
        button = [ItemRighButton buttonWithType:UIButtonTypeCustom];
    }
    [button setTitle:title forState:UIControlStateNormal];
    [button setImage:[UIImage imageNamed:image] forState:UIControlStateNormal];
    [button setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [button addTarget:target action:action forControlEvents:UIControlEventTouchUpInside];
    button.frame = CGRectMake(0, 0, 60, 44);
    button.titleLabel.font = [UIFont systemFontOfSize:13];
    return [[UIBarButtonItem alloc]initWithCustomView:button];
}

@end

File: Example/堆糖/堆糖/Classes/Other(其他)/Extension/UIBarButtonItem+Extension.swift
//
//  UIBarButtonItem+Extension.swift
//  堆糖
//
//  Created by 金亮齐 on 2017/4/13.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

import UIKit

extension UIBarButtonItem {
    
/**创建一个UIBarButtonItem(有图片)*/
convenience init(imageName: String,titleName: String, target: AnyObject?, action: Selector){
    
    let button = UIButton(type: UIButtonType.Custom)
    button.setBackgroundImage(UIImage(named: imageName), forState: UIControlState.Normal)
    button.setBackgroundImage(UIImage(named: imageName), forState: UIControlState.Highlighted)
    button.addTarget(target, action: action, forControlEvents: UIControlEvents.TouchUpInside)
    button.sizeToFit()
    self.init(customView: button)
}
    
/** 创建一个UIBarButtonItem(无图片) */
convenience init(title: String,titleColor: UIColor, target: AnyObject?, action: Selector){
    
    let button = UIButton(type: UIButtonType.Custom)
    button.setTitle(title, forState: UIControlState.Normal)
    button.setTitleColor(titleColor, forState: UIControlState.Normal)
    button.titleLabel?.font = UIFont.systemFontOfSize(15)
    button.addTarget(target, action: action, forControlEvents: UIControlEvents.TouchUpInside)
    button.sizeToFit()
    self.init(customView: button)
    }
}